<launch>
  <arg name="launch_prefix" default="" /> <!-- set to value="gdbserver localhost:10000" for remote debugging -->
  <arg name="node_namespace" default="apriltag_ros_continuous_node" />
  <arg name="mav_name" default="drone" />

  <arg name="camera_type" value="camera_na" />
  <arg name="camera_name" default="$(arg mav_name)/camera_sensor/camera_na" />
  <arg name="camera_frame" default="$(arg mav_name)/camera_sensor/$(arg camera_type)_optical_link" />
  <arg name="image_topic" default="image_rect" />
  <arg name="tag_definition_file" default="tags_rotors.yaml" />

  <!-- Set parameters -->
  <rosparam command="load" file="$(find quad_state_estimation)/config/apriltag_settings_rotors.yaml" ns="$(arg node_namespace)" />
  <rosparam command="load" file="$(find quad_state_estimation)/config/$(arg tag_definition_file)" ns="$(arg node_namespace)" />
  
  <node pkg="apriltag_ros" type="apriltag_ros_continuous_node" name="$(arg node_namespace)" clear_params="true" output="screen" launch-prefix="$(arg launch_prefix)" >
    <!-- Remap topics from those used in code to those on the ROS network -->
    <remap from="image_rect" to="$(arg camera_name)/$(arg image_topic)" />
    <remap from="camera_info" to="$(arg camera_name)/camera_info" />

    <param name="camera_frame" type="str" value="$(arg camera_frame)" />
    <param name="publish_tag_detections_image" type="bool" value="true" />      <!-- default: false -->
  </node>
</launch>
